### PR TITLE
CS0050 code fix

### PR DESCRIPTION
**Pull request by baks**
Fixes `CS0050` in #381 

I added `CS0050` to `InconsistentAccessibilityCodeFixProvider` with unit tests.
Also changed code action messages to use localizable strings.
Common methods used by `InconsistentAccessibilityInMethodParameter/(ReturnType)` were moved as extension methods.

### CODE REVIEW

File: src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodReturnType.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            var result = new InconsistentAccessibilityInfo();
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var methodThatRaisedError = syntaxRoot.FindNode(diagnostic.Location.SourceSpan).DescendantNodesAndSelf().OfType<MethodDeclarationSyntax>().FirstOrDefault();
+            if (methodThatRaisedError != null)
+            {
+                result.TypeToChangeAccessibility = methodThatRaisedError.ReturnType;
+                result.CodeActionMessage = string.Format(CodeActionMessage.ToString(), methodThatRaisedError.ReturnType, methodThatRaisedError.GetIdentifier().ValueText);
+                result.NewAccessibilityModifiers = methodThatRaisedError.CloneAccessibilityModifiers();
+            }
+
```
Comment (giggio) [inline src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodReturnType.cs]: You will have to verify if the return type is in source, the same problem from `CC0051`.

MyDependence.dll:

````csharp
    public class DependedUpon
    {
        protected class A { }
    }
````

My code:

````csharp
    public class Class1 : ClassLibrary6.DependedUpon
    {
        public void Foo()
        {
            var a = new A();
        }
        public A Bar()
        {
            throw new Exception();
        }
    }
````

I can't change the accessibility of `DependedUpon.A` to be public, it is not in my source.

File: src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodParameter.cs [hunk 2]
```diff
@@ -8,11 +8,14 @@
 using System.Threading.Tasks;
 using System.Text.RegularExpressions;
 using System.Globalization;
+using CodeCracker.Properties;
 
 namespace CodeCracker.CSharp.Design.InconsistentAccessibility
 {
     public sealed class InconsistentAccessibilityInMethodParameter : InconsistentAccessibilityInfoProvider
     {
+        private static readonly LocalizableString CodeActionMessage = new LocalizableResourceString(nameof(Resources.InconsistentAccessibilityInMethodParameter_CodeActionMessage), Resources.ResourceManager, typeof(Resources));
```
Comment (giggio) [inline src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodParameter.cs]: The correct name for the resource would be `InconsistentAccessibilityInMethodParameter_Title`.

File: src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodReturnType.cs (hunk 1 above)
Comment (baks) [inline src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodReturnType.cs]: This is done by `IncosistentAccessibilityCodeFixProvider` in `RegisterCodeFixesAsync`

File: src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodParameter.cs (hunk 2 above)
Comment (baks) [inline src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodParameter.cs]: I'll correct that

File: src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodReturnType.cs (hunk 1 above)
Comment (giggio) [inline src/CSharp/CodeCracker/Design/InconsistentAccessibility/InconsistentAccessibilityInMethodReturnType.cs]: Oh, right! Sorry about that.

**Pull request merged**